src/Analyzers/CSharp/Analyzers/ConstantInterpolatedString/CSharpConstantInterpolatedStringAnalyzer.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.CSharp.Shared.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConstantInterpolatedString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)] // constant interpolated strings is a C#-only feature.
+    internal sealed class CSharpConstantInterpolatedStringAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public CSharpConstantInterpolatedStringAnalyzer() : base(
+            IDEDiagnosticIds.UseConstantInterpolatedStringDiagnosticId,
+            EnforceOnBuildValues.UseConstantInterpolatedString,
+            option: null,
+            new LocalizableResourceString(nameof(CSharpAnalyzersResources.Use_constant_interpolated_string), CSharpAnalyzersResources.ResourceManager, typeof(CSharpAnalyzersResources)))
+        {
+        }
+
+        public override DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        protected override void InitializeWorker(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(context =>
+            {
+                if (!((CSharpCompilation)context.Compilation).LanguageVersion.HasConstantInterpolatedStrings())
+                {
+                    return;
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    var operation = (IBinaryOperation)context.Operation;
+                    if (!ShouldAnalyze(operation))
+                    {
+                        return;
+                    }
+
+                    if (!AllOperandsAreStringLiterals(operation))
```

Comment:
AllOperandsAreStringConstants?

Reply:
@CyrusNajmabadi `nameof(C)` is a string constant, but not a literal. The function checks for literals specifically.

But I'm feeling I could remove this check altogether (or if it's problematic, relex it a bit)? What do you think?

Reply:
I think it would make sense that if it's all constant strings that we offer this :-)

Reply:
Ok now I remembered why I did it this way. If all operands are really "literals", then a message saying "Use constant interpolated string" doesn't make much sense, there is no need for interpolation. It just needs to be a single string (e.g, `"Hello" + "World"` → `"HelloWorld"` which is different than `"Hello" + nameof(World)` → `$"Hello{nameof(World)}"`).

I could change the analyzer name to be "Use a single string" and support this scenario, or do you have a better idea?